tests/test_homogeneous_collections.py

```diff
@@ -121,11 +121,93 @@ class RootModel(BaseXmlModel, tag='model'):
     assert_xml_equal(actual_xml, xml)
 
 
-def test_homogeneous_definition_errors():
-    with pytest.raises(errors.ModelFieldError):
-        class TestModel(BaseXmlModel):
-            attr1: List[int] = attr()
+def test_text_list_extraction():
+    class RootModel(BaseXmlModel, tag="model"):
+        values: List[int]
+
+    xml = '''
+    <model>1 2 70 -34</model>
+    '''
+
+    actual_obj = RootModel.from_xml(xml)
+    expected_obj = RootModel(
+        values = [1, 2, 70, -34],
+    )
+
+    assert actual_obj == expected_obj
+
+    actual_xml = actual_obj.to_xml()
+    assert_xml_equal(actual_xml, xml)
+
+
+def test_text_tuple_extraction():
+    class RootModel(BaseXmlModel, tag="model"):
+        values: Tuple[int, ...]
 
+    xml = '''
+    <model>1 2 70 -34</model>
+    '''
+
+    actual_obj = RootModel.from_xml(xml)
+    expected_obj = RootModel(
+        values=[1, 2, 70, -34],
+    )
+
+    assert actual_obj == expected_obj
+
+    actual_xml = actual_obj.to_xml()
+    assert_xml_equal(actual_xml, xml)
+
+
+def test_attr_list_extraction():
+    class RootModel(BaseXmlModel, tag="model"):
+        values: List[float] = attr()
+
+    xml = '''
+    <model values="3.14 -1.0 300.0"/>
```

Comment:
Will there be support for custom delimiters for attribute values? The space char seems to be the default for now, but curious if other delimiters will be supported in the future. I can imagine values might be separated by other chars like `|` and `,` .
```xml
# comma delimiter
<model values="3.14,-1.0,300.0"/> 

# vertical bar delimiter.
<model values="3.14|-1.0|300.0"/>
```

Reply:
The current PR implements support for XML Schema lists, https://www.w3.org/TR/xmlschema11-2/#atomic-vs-list , which, afaict, only support white-space delimited values. I have no plans to implement other delimiters as this covers my (or rather my project $work's) needs. For other kinds of delimiters, I think you're better off using elements, as they are more general.

However, if someone need to consume lists with other delimiters, I'd gladly help with the implementation

Reply:
Hey @ajoino, thank you for your reply. Yes the main issue I have is on the consumption side. At the moment my challenge is with the deserialization of the XML. I am hoping to be able to parse the `pageReferences` attribute and represent it as type `List[int]` instead of `str`. However, compared to your space delimited examples in this PR, I dealing with both `|` and `,` delimiters.

**Example XML:**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<books>
    <book id="1" name="BookA" pageReferences="2,6,14"/>
    <book id="2" name="BookB" pageReferences="1,8,57"/>
</books>
```
**Python Classes:**
```python
class Book(BaseXmlModel, tag="book"):
    id: str = attr(name="id")
    name: str = attr(name="name")
    page_references: str = attr(name="pageReferences")


class BookResponse(BaseXmlModel, tag="books"):
    books: Optional[List[Book]] = element()
```

I was mainly wondering if it would be possible to pass a custom delimiter param to `attr()`. This could act as an override to the default space delimiter whenever it is specified.
```python
class Book(BaseXmlModel, tag="book"):
    ...
    page_references: List[int] = attr(name="pageReferences",  delimiter=",")
```